examples/campaign_management/update_campaign_criterion_bid_modifier.py

```diff
@@ -0,0 +1,79 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Sets a bid modifier for the mobile platform on the given campaign."""
+
+import argparse
+import sys
+
+import google.ads.google_ads.client
+from google.api_core import protobuf_helpers
+
+
+_BID_MODIFIER = 1.5
+
+
+def main(client, customer_id, criterion_id):
+    campaign_criterion_service = client.get_service(
+        'CampaignCriterionService', version='v2')
+
+    criterion_rname = campaign_criterion_service.campaign_criteria_path(
+        customer_id, criterion_id)
+
+    campaign_criterion_operation = client.get_type(
+        'CampaignCriterionOperation', version='v2')
+    campaign_criterion = campaign_criterion_operation.update
+    campaign_criterion.resource_name = criterion_rname
+    campaign_criterion.bid_modifier.value = _BID_MODIFIER
+    campaign_criterion.device.type = client.get_type(
+        'DeviceEnum', version='v2').MOBILE
+    fm = protobuf_helpers.field_mask(None, campaign_criterion)
+    campaign_criterion_operation.update_mask.CopyFrom(fm)
+
+    try:
+        campaign_criterion_response = (
+            campaign_criterion_service.mutate_campaign_criteria(
+                customer_id, [campaign_criterion_operation]))
+    except google.ads.google_ads.errors.GoogleAdsException as ex:
+        print(f'Request with ID "{ex.request_id}" failed with status '
+              f'"{ex.error.code().name}" and includes the following errors:')
+        for error in ex.failure.errors:
+            print(f'\tError with message "{error.message}".')
+            if error.location:
+                for field_path_element in error.location.field_path_elements:
+                    print(f'\t\tOn field: {field_path_element.field_name}')
+        sys.exit(1)
+
+    print('Campaign criterion with resource name '
+          f'"{campaign_criterion_response.results[0].resource_name}" was '
+          'modified.')
+
+
+if __name__ == '__main__':
+    # GoogleAdsClient will read the google-ads.yaml configuration file in the
+    # home directory if none is specified.
+    google_ads_client = (google.ads.google_ads.client.GoogleAdsClient
+                         .load_from_storage())
+
+    parser = argparse.ArgumentParser(
+        description=('Updates the bid modifier and device type for the given '
+                     'customer ID and campaign criterion ID.'))
+    # The following argument(s) should be provided to run the example.
+    parser.add_argument('-c', '--customer_id', type=str,
+                        required=True, help='The Google Ads customer ID.')
+    parser.add_argument('--criterion_id', type=str, required=True,
+                        help='The criterion ID.')
```

Comment:
missing campaign id argument

Reply:
Josh also asked to make bid_modifier an argument as well (though it isn't in the golden yet)

Reply:
I actually don't get why we want to skip campaign ID here.
It's true that campaign criterion ID includes campaign ID, but it's inconvenient for users to create the ID by themselves.
(That's why we have utilities to create a resource name for ad group criterion, campaign criterion, etc.)

For this one, please follow what's in Java PR (https://github.com/googleads/google-ads-java/pull/232/files), i.e., separate campaign ID and criterion ID. This will be consistent with other examples too.

Reply:
Strictly speaking, the `$CAMPAIGN_ID~$RAND_ID` value *is* the campaignCriteria's resource ID. If we wanted to better distinguish that the criterion is hierarchically a sub-resource of its associated campaign, shouldn't the resource name be structured as follows:

`customers/$CUSTOMER_ID/campaign/$CAMPAIGN_ID/campaignCriteria/$CAMPAIGN_CRITERIA_ID`

In any case, I will follow the recommendations per your email.

Reply:
- It's actually not always RAND_ID, especially for many the campaign criteria.
- I believe there were some (long) discussions about this before. However, I don't have all contexts to discuss with you. It probably will not be changed in the long run, so let's ignore this for now, and follow my recommendations to meet consistency among files and libraries.